### DISCUSSION TITLE
docs/how-it-works/

### DISCUSSION THREAD

**Discussion by giscus** [casbin/casbin#1353] [Docs comments]
# docs/how-it-works/

How Casbin Works

https://casbin.org/docs/how-it-works/

**Comment by mbesurich** [comment]
I want to use some(where (p.eft == allow)) && !some(where (p.eft == deny)) with superadmin. And I get false for every requests regardless the policies, I can have only 1 policy with deny effect, and this policy does not match to any existing endpoint. 
Is it possible to use e = some(where (p.eft == allow)) && !some(where (p.eft == deny)) and superadmin in same model?

**Comment by yujiachen-y** [comment]
'jCasbin' means 'java casbin'?

**Reply by hsluoyz** [reply]
Yes, see: https://github.com/casbin/jcasbin

**Comment by yujiachen-y** [comment]
> But you **SHOULD** make sure that the length of the array is **MORE** than 1, otherwise it will cause a panic.

In this sentence, better replace 'SHOULD' by 'MUST'.

**Reply by hsluoyz** [reply]
Feel free to fix it by making PR to: https://github.com/casbin/casbin-website-v2
